File: Selenium/Activities/Activity18.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import java.util.List;

public class Activity18 {
    public static void main(String[] args) throws InterruptedException {

        // Initialise Webdriver
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");

        // Launch the GUI and Get the Title
        String title = driver.getTitle();
        System.out.print("Title of the Page is : " + title + "\n");

        // Get the Multi Select Options
        WebElement optionSelect = driver.findElement(By.id("multi-select"));
        Select multiSelect = new Select(optionSelect);

        // Select items
        multiSelect.selectByVisibleText("Javascript");
        multiSelect.selectByIndex(4);
        multiSelect.selectByIndex(5);
        multiSelect.selectByIndex(6);
        multiSelect.selectByValue("node");

        // Deselect item @ 5th index
        multiSelect.deselectByIndex(5);

        //Wait for 3 Seconds and Close the Browser
        Thread.sleep(3000);
        driver.quit();
    }
}

File: Selenium/Activities/Activity18.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/selects")
    home_title = driver.title
    print("Title of the Home Page is : {}".format(home_title))

    # Get the Multi Select Options
    optionSelect = driver.find_element(By.ID, "multi-select")
    multiSelect = Select(optionSelect)

    # Select items
    multiSelect.select_by_visible_text("Javascript")
    multiSelect.select_by_index(4)
    multiSelect.select_by_index(5)
    multiSelect.select_by_index(6)
    multiSelect.select_by_value("node")

    # Deselect item @ 5th index
    multiSelect.deselect_by_index(5)

    # Wait for 3 Seconds and Close the Browser
    time.sleep(3)
    driver.close()
